### PR TITLE
mongodb26: 2.6.12

### PR DESCRIPTION
**Pull request by AntiFish03**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-versions/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
- [x] Does your submission adhere to Versions "Acceptable Formulae" [guidelines](https://github.com/Homebrew/homebrew-versions/blob/07b17dfb9f1/README.md#acceptable-formulae)?

-----
Updating mongodb version to 2.6.12 from 2.6.11
Adding macOS Sierra compatibility

### CODE REVIEW

File: mongodb26.rb [hunk 1]
```diff
@@ -22,8 +22,8 @@ class Mongodb26 < Formula
   # This modifies the SConstruct file to include 10.10 and 10.11 as accepted build options.
```
Comment (MikeMcQuaid) [inline mongodb26.rb]: Could you update this comment accordingly? Also, replacing the patch with an `inreplace` would be ideal.

Comment (AntiFish03) [inline mongodb26.rb]: Request changes have been made.

File: mongodb26.rb [hunk 2]
```diff
@@ -18,16 +19,10 @@ class Mongodb26 < Formula
...
-  if MacOS.version >= :yosemite
-    patch do
-      url "https://gist.githubusercontent.com/asmarques/4e4e66121dbe46a08933/raw/d6753551ff3849bf0fb5ca87c0a96e1255f4254d/mongodb26-elcapitan.diff"
-      sha256 "2f79588a8a15660908d8d071655541c9ab0ac425550b02dc454c861c75f86606"
-    end
-  end
-
   def install
+    # This modifies the SConstruct file to include additional osx versions as accepted build options.
+    inreplace "SConstruct", /osx_version_choices = \[.+?\]/, "osx_version_choices = #{OSX_VERSION_CHOICES.inspect}"
```
Comment (MikeMcQuaid) [inline mongodb26.rb]: Either hard-code the constant values or just use `MacOS.version` here to always make it the current version.

Comment (AntiFish03) [inline mongodb26.rb]: Hard coded as requested. With updated comment

**Pull request closed without merging**